pkg/controller/deployment/deployment_controller.go

```diff
@@ -1200,12 +1222,18 @@ func (dc *DeploymentController) cleanupOldReplicaSets(oldRSs []*extensions.Repli
 	return utilerrors.NewAggregate(errList)
 }
 
-func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, deployment *extensions.Deployment) error {
+func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, deployment *extensions.Deployment, cond *extensions.DeploymentCondition) error {
 	totalActualReplicas, updatedReplicas, availableReplicas, unavailableReplicas, err := dc.calculateStatus(allRSs, newRS, deployment)
 	if err != nil {
 		return err
 	}
 	newDeployment := *deployment
+	if cond != nil {
+		if newDeployment.Status.Conditions == nil {
+			newDeployment.Status.Conditions = []extensions.DeploymentCondition{}
```

Comment:
This clears the conditions every sync? Would the ScaleUp/ScaleDown conditions typically be observable?